document_guides/requirement_guide.md

```diff
@@ -1,19 +1,42 @@
 # Requirement Guide
 
-In the `documents` folder of each project, there is a `reqirement.md` specifying the business requirment of the project.
+In the `docs` folder of each project, there is a `reqirement.md` specifying the
+business requirment of the project.
 
 the requirement should contains the fellow content:
 
-### The purpose of this project
-we should describe the purpose of this project, why we start this project.
- 
-### The output of this project
-this may be a jar file, jar file, docker image etc. 
+## 1. Introduction 
+a short description about project 
+
+## 2. Purpose
+
+1. describe the purpose of project
+2. what kinds of services or features this project could provide 
+
+Taking `category` service project for example, it could provide such sorts of
+features:
++ create category
++ delete category
++ update category
++ get individual category
++ get all categories
+
+Thus, you should explicitly point out the supported features of your project as
+example show
+
+## 3. Requirement
+it is mainly about technical requirement which pertains to the technical aspects
+that project should fulfill. This requirement typically refer to how project is
+built, such as what kind of standard it should meet, Taking `category` project
+for
+example,
+[category requirement](https://github.com/reactivesw/category/tree/master/docs)
+demonstrates the basic requirement about category, such as parameters whether it
+is required or not, relation between different category. Then it shows the
+different requirement for features it provides. 
+
 
-### The king of service provided
-Rest API, grpc api, or web content etc.
 
-### The detail of the service or content that this project provided
 
 
```

Comment:
My opinion:

# Title --- usually the service name
*description of this micro-service* --- the purpose of this service
# The Services or futures provided
*services provided*
# Key points
*key points of the service*, for example: `password must be at least 8 characters`

# Other services needed, and why.